extensions/jdbc/jdbc-mysql/deployment/src/main/java/io/quarkus/jdbc/mysql/deployment/JDBCMySQLProcessor.java

```diff
@@ -86,6 +78,16 @@ NativeImageEnableAllTimeZonesBuildItem enableAllTimeZones() {
         return new NativeImageEnableAllTimeZonesBuildItem();
     }
 
+    @BuildStep
+    NativeImageSystemPropertyBuildItem disableAbandonedConnectionCleanUpInNativeMode() {
+        return new NativeImageSystemPropertyBuildItem("com.mysql.cj.disableAbandonedConnectionCleanup", "true");
+    }
+
+    @BuildStep
```

Comment:
```suggestion
    @BuildStep(onlyIfNot = NativeBuild.class)
```
would make it clearer.

Reply:
I was wondering about that.. woudldn't it be enough to just have this single buildStep?

Reply:
> I was wondering about that.. woudldn't it be enough to just have this single buildStep?

Would `SystemPropertyBuildItem` be consumed and produce the `NativeImagePropertyBuildItem`? My first time producing either of them so I might be missing something.

Reply:
I don't know :) @gsmet ?

Reply:
I think it's OK the way it is now.

Reply:
> > I was wondering about that.. woudldn't it be enough to just have this single buildStep?

Tested it and it works by only producing the `SystemPropertyBuildItem`. But to get an answer to my question below ... 

> Would `SystemPropertyBuildItem` be consumed and produce the `NativeImageSystemPropertyBuildItem`? My first time producing either of them so I might be missing something.

With only the `SystemPropertyBuildItem` the command to generate native image is:

```bash
native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-DCoordinatorEnvironmentBean.transactionStatusManagerEnable=false -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=en -J-Dfile.encoding=UTF-8 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -jar quarkus-integration-test-jpa-mysql-999-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:+AddAllCharsets -H:EnableURLProtocols=http,https --enable-all-security-services -H:NativeLinkerOption=-no-pie --no-server -H:-UseServiceLoaderFeature -H:+StackTrace quarkus-integration-test-jpa-mysql-999-SNAPSHOT-runner
```

With the `NativeImageSystemPropertyBuildItem` the command become:
```bash
native-image -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dcom.mysql.cj.disableAbandonedConnectionCleanup=true -J-DCoordinatorEnvironmentBean.transactionStatusManagerEnable=false -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=en -J-Dfile.encoding=UTF-8 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -jar quarkus-integration-test-jpa-mysql-999-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:+AddAllCharsets -H:EnableURLProtocols=http,https --enable-all-security-services -H:NativeLinkerOption=-no-pie --no-server -H:-UseServiceLoaderFeature -H:+StackTrace quarkus-integration-test-jpa-mysql-999-SNAPSHOT-runner
```

Notice the `-J-Dcom.mysql.cj.disableAbandonedConnectionCleanup=true` in the second command which is missing from the first.

I think for the sake of being explicit in the generated command, I'd stick with producing the two build items.

Reply:
Sure. But thanks for testing it! Good to know.